plugins/modules/swim_workflow_manager.py

```diff
@@ -810,24 +810,30 @@ def get_device_id(self, params):
             no device or multiple devices match the criteria, it raises an exception.
         """
         device_id = None
-        response = self.dnac._exec(
-            family="devices",
-            function='get_device_list',
-            op_modifies=True,
-            params=params,
-        )
-        self.log("Received API response from 'get_device_list': {0}".format(str(response)), "DEBUG")
+        self.log(params)
+        try:
+            response = self.dnac._exec(
+                family="devices",
+                function='get_device_list',
+                op_modifies=True,
+                params=params,
+            )
+            self.log("Received API response from 'get_device_list': {0}".format(str(response)), "DEBUG")
 
-        device_list = response.get("response")
+            device_list = response.get("response")
 
-        if (len(device_list) == 1):
-            device_id = device_list[0].get("id")
-            self.log("Device Id: {0}".format(str(device_id)), "INFO")
-        else:
-            self.msg = "Device with params: '{0}' not found in Cisco Catalyst Center so can't fetch the device id".format(str(params))
-            self.log(self.msg, "WARNING")
+            if (len(device_list) == 1):
```

Comment:
if device_list is None:
            self.log("Device list is empty; no devices found for given parameters.", "WARNING")
            raise ValueError("No devices found")

        if len(device_list) == 1:
            device_id = device_list[0].get("id")
            self.log("Successfully retrieved device ID: {0}".format(device_id), "INFO")
            return device_id
        
        self.log("Multiple devices found for parameters: {0}".format(params), "ERROR")
        raise ValueError("Multiple devices found")
        
        
    except ValueError as ve:
        self.msg = "Error: {0}. Unable to fetch unique device ID with parameters: {1}".format(str(ve), params)
        self.status = "failed"
        self.log(self.msg, "ERROR")
        self.result['response'] = self.msg
        return None

    except Exception as e:
        self.msg = "An unexpected error occurred while retrieving device ID: {0}".format(str(e))
        self.status = "failed"
        self.log(self.msg, "ERROR")
        self.result['response'] = self.msg
        return None

Reply:
updated the code